pennylane_sf/expectations.py

```diff
@@ -94,6 +94,23 @@ def mean_photon(state, wires, params):
     return state.mean_photon(wires[0])
 
 
+def number_expectation(state, wires, params):
+    """Computes the expectation value of tensor products consisting of the
+    ``qml.NumberOperator`` observable on specified modes in Strawberry Fields.
+
+    Args:
+        state (strawberryfields.backends.states.BaseState): the quantum state
+        wires (Sequence[int]): the sequence of modes to measure
+        params (Sequence): sequence of parameters (not used)
+
+    Returns:
+        float, float: the expectation value of the number operator and its
+            variance
+    """
+    # pylint: disable=unused-argument
+    return state.number_expectation(wires), None
```

Comment:
@antalszava, did you want to add an if statement here?
```python
if len(wires) == 1:
    return mean_photon(state, wires, params)
return state.number_expectation(wires), None
```

The only difference will be that if it's only acting on 1 mode, variance will be supported. But this is minor.

Reply:
Right, yes, this is a great idea! The reason why I haven't added it, is because by this step of the execution it has already been taken care of. As [`TensorN` for `len(wires) == 1` returns the  `NumberOperator` ](https://github.com/XanaduAI/pennylane/blob/bcc4de1b2af476ed0e05ea0ded54ab57a3cd3754/pennylane/ops/cv.py#L860), upon instantiation of `TensorN` we get the `NumberOperator` and then here in PennyLane-SF we already deal with a `NumberOperator` instance. Therefore this function is called when `len(wires) != 1`.

Reply:
Oh, good point! Disregard my comment in that case, I had forgotten we had made that decision in PennyLane core :)